ElDorito/Source/Modules/ModuleGame.cpp

```diff
@@ -1056,6 +1063,57 @@ namespace
 
 	}
 
+	bool CommandDownloadVariant(const std::vector<std::string>& Arguments, std::string& returnInfo)
+	{
+		if (Arguments.size() < 1)
+			return false;
+		returnInfo = "Trying to download:" + Arguments[0] + "\n";
+		HttpRequest req(L"ElDewrito/" + Utils::String::WidenString(Utils::Version::GetVersionString()), L"", L"");
+		if (!req.SendRequest(Utils::String::WidenString(Arguments[0]), L"GET", L"", L"", L"", NULL, 0))
+		{
+			returnInfo += "Error fetching variant file. Error code: " + std::to_string(req.lastError);
+			return false;
+		}
+
+		// mapv = Map Variant, mpvr = Game Mode (Multiplayer) Variant
+		std::string variantFormat = std::string(&req.responseBody.at(312), &req.responseBody.at(316));
+		returnInfo += "Variant Format Header (" + variantFormat + ")\n";
+
+		if (req.responseBody.size() != 0x1000 && req.responseBody.size() != 0xF000)
+		{
+			returnInfo += "Incorrect variant file size (" + std::to_string(req.responseBody.size()) + " bytes)";
+			return false;
+		}
+		returnInfo += "Response variant file size (" + std::to_string(req.responseBody.size()) + " bytes)\n";
+		std::string header = std::string(&req.responseBody.at(0x0), &req.responseBody.at(0x4));
+		bool valid = (header == "_blf");
+
+		BYTE variant_name[0x20];
+		// Default entire array to 0
+		std::fill_n(variant_name, 0x20, 0);
+
+		for (int i = 0; i <= 31; i++)
+			variant_name[i] = req.responseBody[i + 72];
+
+		if (valid) {
+			auto game = &Modules::ModuleGame::Instance();
+			std::wstring_convert<std::codecvt_utf8<wchar_t>> wstring_to_string;
+
+			game->variant_vector = req.responseBody;
```

Comment:
wouldn't something like `variant_data` or `variant_data_vector` be a better name?